{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "dbf145bd-31a3-449f-aefc-7c0e892d57dc",
   "metadata": {},
   "source": [
    "# **GATK Somatic Mutation Calling on Breast Cancer Cell Line HCC1143 (BAM → MAF).**\n",
    "\n",
    "## Dataset \n",
    "\n",
    "In this analysis, we process WES data from the well-characterized breast cancer cell line HCC1143 and matched normal sample. This pair of samples is widely used in benchmarking and demonstration workflows and is supported by publicly available reference results, enabling comparison and validation of somatic variant calling outputs. \n",
    "\n",
    "To reduce computational complexity the analysis is restricted to **chromosome 17**, which harbors several clinically relevant genes (e.g. TP53).\n",
    "\n",
    "## Data source \n",
    "\n",
    "Input data and reference resources were obtained from official Broad Institute repositories:\n",
    "\n",
    "**Germline population resource (gnomAD, ExAC):**\n",
    "\n",
    "https://console.cloud.google.com/storage/browser/gatk-best-practices/somatic-b38\n",
    "\n",
    "Files used: af-only-gnomad.hg38.vcf.gz, af-only-gnomad.hg38.vcf.gz.tbi, small_exac_common_3.hg38.vcf.gz, small_exac_common_3.hg38.vcf.gz.tbi\n",
    "\n",
    "**WES BAM files (tumor and matched normal):**\n",
    "\n",
    "https://console.cloud.google.com/storage/browser/gatk-tutorials/workshop_1903/3-somatic\n",
    "\n",
    "Files used: bams/tumor.bam,tumor.bai, normal.bam, normal.bai., refs/Homo_sapiens_assembly38.fasta, refs/Homo_sapiens_assembly38.fasta.fai\n",
    "\n",
    "**Curated mutation data for HCC1143 from the DepMap portal:**\n",
    "\n",
    "https://depmap.org/portal/cell_line/ACH-000374?tab=mutations\n",
    "\n",
    "Files used: HCC1143 mutations.csv\n",
    "\n",
    "All data are aligned to the GRCh38 / hg38 reference genome. \n",
    "\n",
    "## Objective\n",
    "\n",
    "The objective of this notebook is to perform somatic variant calling on the HCC1143 tumor–normal WES dataset. \n",
    "\n",
    "## Workflow\n",
    "\n",
    "The analysis follows the GATK Best Practices somatic variant calling workflow:\n",
    "\n",
    "**BAM files → Mutect2 → Variant filtering → Functional annotation (MAF) → Comparison with known HCC1143 mutations**\n",
    "\n",
    "All variant calling steps were performed using GATK v4.6.2.0 executed via the official Broad Institute Docker image."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8aa56382-31b2-4817-96f0-33860b9b0da4",
   "metadata": {},
   "source": [
    "## Somatic variant calling with Mutect2\n",
    "\n",
    "The following command runs Mutect2 inside a GATK Docker container, using matched tumor and normal samples. \n",
    "\n",
    "Variant calling was restricted to chromosome 17. \n",
    "\n",
    "In addition to generating a somatic VCF, a BAM file containing Mutect2 realigned reads is produced for visualization and quality assessment in IGV.\n",
    "\n",
    "A population allele-frequency resource derived from gnomAD was used to reduce false-positive calls from germline variants. The parameters --max-population-af and --af-of-alleles-not-in-resource define conservative population-frequency priors in Mutect2 and are consistent with GATK Best Practices for paired tumor–normal somatic variant calling."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "471c7fb6-373c-4f61-98c5-cfbebb5006b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "! gatk Mutect2 \\\n",
    "    -R /data/ref/Homo_sapiens_assembly38.fasta \\\n",
    "    -I /data/bams/tumor.bam \\\n",
    "    -I /data/bams/normal.bam \\\n",
    "    -tumor HCC1143_tumor \\\n",
    "    -normal HCC1143_normal \\\n",
    "    --germline-resource af-only-gnomad.hg38.vcf.gz \\\n",
    "    --af-of-alleles-not-in-resource 0.01 \\\n",
    "    --max-population-af 0.01 \\\n",
    "    -L chr17 \\\n",
    "    -O /data/somatic.chr17.vcf.gz \\\n",
    "    -bamout /data/mutect2_tumor_normal.chr17.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23cd84c3-91ef-4a6e-abfd-0780db8dc874",
   "metadata": {},
   "source": [
    "GnomAD filtering reduces false-positive somatic calls from common germline variants but does not account for cross-sample contamination. Therefore, the next step is contamination estimation."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d568b22d-da06-4914-bfc6-c41b7b123600",
   "metadata": {},
   "source": [
    "## Сontamination estimation\n",
    "\n",
    "Sample contamination was estimated using GetPileupSummaries and CalculateContamination based on common germline variants in the tumor and matched normal sample to improve specificity of downstream filtering.\n",
    "\n",
    "GetPileupSummaries is performed independently for the tumor and the matched normal samples, based on a curated set of common germline SNPs (file small_exac_common_3.hg38.vcf.gz). GetPileupSummaries produces per-sample allele count summary tables. \n",
    "\n",
    "Then CalculateContamination compares observed allele fractions in the tumor to those expected based on the normal sample, identifies systematic VAF shifts at common germline SNPs and infers the most likely contamination fraction that explains these shifts. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2eee96d2-e891-4fae-8a9f-c67d46ee9b4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gatk GetPileupSummaries \\\n",
    "    -I bams/tumor.bam \\\n",
    "    -V small_exac_common_3.hg38.vcf.gz \\\n",
    "    -L small_exac_common_3.hg38.vcf.gz \\\n",
    "    -O tumor_summary.table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e545738-a7ea-4270-8eaf-c079d25857be",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gatk GetPileupSummaries \\\n",
    "    -I bams/normal.bam \\\n",
    "    -V somatic-hg38-small_exac_common_3.hg38.vcf.gz \\\n",
    "    -L somatic-hg38-small_exac_common_3.hg38.vcf.gz \\\n",
    "    -O normal_summary.table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65841b88-de8a-4dce-a36c-9a722de9952d",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gatk CalculateContamination \\\n",
    "    -I tumor_summary.table \\\n",
    "    -matched normal_summary.table \\\n",
    "    -O contamination.table "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7bead14e-50c6-44b2-af06-1dbe697fa5a1",
   "metadata": {},
   "source": [
    "Inspecting the contamination.table, we observe an estimated contamination level of approximately ~1.1%, indicating low admixture in the tumor sample."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "f4cb61ce-db2a-4437-8950-a0d2d5da7f5f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample\tcontamination\terror\n",
      "HCC1143_tumor\t0.011485364960150258\t0.0021979540554002835\n"
     ]
    }
   ],
   "source": [
    "!cat contamination.table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46f05f92-a135-4621-ab44-7082c2cc4ccd",
   "metadata": {},
   "source": [
    "##  Somatic variant filtering with FilterMutectCalls\n",
    "\n",
    "In the next step, candidate somatic variants produced by Mutect2 assigns filter statuses using FilterMutectCalls (e.g. PASS, contamination,  technical artifact, low evidence and other flags). FilterMutectCalls applies a probabilistic model to integrate multiple sources of evidence, including sequencing depth, base and mapping quality, contamination, strand bias, and local haplotype context. \n",
    "\n",
    "The resulting VCF retains all candidate variants but provides explicit annotations that facilitate informed downstream selection of high-confidence somatic calls, rather than producing a pre-filtered variant set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eddfbde6-b932-4fa0-87c4-6cc3876b2492",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gatk FilterMutectCalls \\\n",
    "    -V somatic.chr17.vcf.gz \\\n",
    "    -R /data/ref/Homo_sapiens_assembly38.fasta \\\n",
    "    --contamination-table contamination.table \\\n",
    "    --stats somatic.chr17.vcf.gz.stats \\\n",
    "    -O somatic_postfilter.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e49b92d5-ef14-4047-8b3b-7d6ef3ef02f0",
   "metadata": {},
   "source": [
    "## Functional annotation with Funcotator (VCF → MAF conversion)\n",
    "\n",
    "Prior to annotation, the required Funcotator data sources for somatic variant analysis on the hg38 reference genome were downloaded using FuncotatorDataSourceDownloader. These data sources include gene models, transcript annotations, and multiple functional annotation tracks used to interpret variant consequences from COSMIC, ClinVar, Uniprot.\n",
    "\n",
    "Functional annotation was then performed with Funcotator on the subset of somatic variants that passed FilterMutectCalls (flag \"PASS\"). Annotated variants were exported in both VCF and MAF formats - the VCF output preserves the complete technical variant representation and the MAF output provides a user-friendly tabular summary optimized for downstream interpretation and comparison with external datasets. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e01c0d5-0c50-46aa-a494-1d72128d640c",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gatk FuncotatorDataSourceDownloader \\\n",
    "  --somatic \\\n",
    "  --hg38 \\\n",
    "  --output funcotator_dataSources \\\n",
    "  --extract-after-download true"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8cd1974-e1d4-423d-8907-a4964095cc8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools view -f PASS somatic_postfilter.vcf.gz -Oz -o somatic_PASS.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00e3ab2a-1777-415b-af4a-feaabd29889b",
   "metadata": {},
   "outputs": [],
   "source": [
    "!bcftools index somatic_PASS.vcf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24cac7d5-fe0d-4ec1-bea5-b5b87ad90157",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gatk Funcotator \\\n",
    "  --data-sources-path funcotator_dataSources.v1.8.hg38.20230908s \\\n",
    "  --ref-version hg38 \\\n",
    "  -R ref/Homo_sapiens_assembly38.fasta \\\n",
    "  -V somatic_PASS.vcf.gz \\\n",
    "  -O somatic_annot.vcf.gz \\\n",
    "  --output-file-format VCF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0639c73-756c-44e7-a94c-9e6b00e847c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "!gatk Funcotator \\\n",
    "  --data-sources-path funcotator_dataSources.v1.8.hg38.20230908s \\\n",
    "  --ref-version hg38 \\\n",
    "  -R ref/Homo_sapiens_assembly38.fasta \\\n",
    "  -V somatic_PASSed.vcf.gz \\\n",
    "  -O somatic_annot.maf \\\n",
    "  --output-file-format MAF"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2baade0-936e-463b-ae6f-752dae2f4a81",
   "metadata": {},
   "source": [
    "## Сomparison with reference data for HCC1143"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cbf3dc01-3721-4afb-91ce-b368702a1f74",
   "metadata": {},
   "source": [
    "The annotated MAF file was loaded into a pandas DataFrame to enable downstream filtering and analysis. Relevant annotation fields were selected, including gene symbol, genomic coordinates, variant classification and type, alleles, codon and protein changes, functional annotations, and selected quality metrics (Tumor LOD). We focus on missense mutations in chromosome 17 gene-coding regions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41ed07be-726f-4ab3-bb63-1be01f1ee2b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "b67f1454-0fe8-4cb9-b37a-5f272a721311",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "muts_raw = pd.read_csv('somatic_annot.maf',sep=\"\\t\",comment=\"#\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e17f1961-830b-4b84-820b-6c89962a9a47",
   "metadata": {},
   "outputs": [],
   "source": [
    "col_selected=['Hugo_Symbol','Chromosome', 'Start_Position','End_Position', 'Variant_Classification', 'Variant_Type',\n",
    "      'Reference_Allele', 'Tumor_Seq_Allele2','Tumor_Seq_Allele1', 'Codon_Change','TLOD', \n",
    "      'Protein_Change', 'GO_Molecular_Function', 'COSMIC_tissue_types_affected']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "1f7c6f4d-78d9-45a6-8a1a-fe0fd3148c82",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "muts_selected=muts_raw[col_selected]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "0c2b2110-fece-44ad-8410-b585b6b7ac51",
   "metadata": {},
   "outputs": [],
   "source": [
    "muts_selected = muts_selected[muts_selected['Variant_Classification'] == 'Missense_Mutation']\n",
    "muts_selected = muts_selected.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "47dce377-a592-4abe-8193-407aa5fa7d09",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Codon_Change</th>\n",
       "      <th>TLOD</th>\n",
       "      <th>Protein_Change</th>\n",
       "      <th>GO_Molecular_Function</th>\n",
       "      <th>COSMIC_tissue_types_affected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NLRP1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>5541887</td>\n",
       "      <td>5541887</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>c.(2668-2670)aGa&gt;aAa</td>\n",
       "      <td>209.99</td>\n",
       "      <td>p.R890K</td>\n",
       "      <td>ATP binding (GO:0005524)|cysteine-type endopep...</td>\n",
       "      <td>biliary_tract(2)|breast(32)|central_nervous_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FBXO39</td>\n",
       "      <td>chr17</td>\n",
       "      <td>6779878</td>\n",
       "      <td>6779878</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>c.(10-12)Gaa&gt;Aaa</td>\n",
       "      <td>37.12</td>\n",
       "      <td>p.E4K</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biliary_tract(2)|breast(11)|central_nervous_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7674220</td>\n",
       "      <td>7674220</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>c.(742-744)cGg&gt;cAg</td>\n",
       "      <td>264.57</td>\n",
       "      <td>p.R248Q</td>\n",
       "      <td>ATP binding (GO:0005524)|chaperone binding (GO...</td>\n",
       "      <td>NS(1495)|adrenal_gland(724)|autonomic_ganglia(...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ANKRD13B</td>\n",
       "      <td>chr17</td>\n",
       "      <td>29609163</td>\n",
       "      <td>29609163</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>c.(643-645)Gac&gt;Cac</td>\n",
       "      <td>22.18</td>\n",
       "      <td>p.D215H</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biliary_tract(2)|breast(11)|central_nervous_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIG3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>34983021</td>\n",
       "      <td>34983021</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.(16-18)Aag&gt;Gag</td>\n",
       "      <td>14.24</td>\n",
       "      <td>p.K6E</td>\n",
       "      <td>ATP binding (GO:0005524)|DNA binding (GO:00036...</td>\n",
       "      <td>biliary_tract(2)|breast(223)|central_nervous_s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>chr17</td>\n",
       "      <td>69016308</td>\n",
       "      <td>69016308</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>c.(2983-2985)cTt&gt;cCt</td>\n",
       "      <td>49.96</td>\n",
       "      <td>p.L995P</td>\n",
       "      <td>ATP binding (GO:0005524)|ATPase activity (GO:0...</td>\n",
       "      <td>biliary_tract(2)|breast(11)|central_nervous_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GGA3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>75239434</td>\n",
       "      <td>75239434</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>c.(1504-1506)cCg&gt;cTg</td>\n",
       "      <td>178.91</td>\n",
       "      <td>p.P502L</td>\n",
       "      <td>ADP-ribosylation factor binding (GO:0030306)</td>\n",
       "      <td>biliary_tract(2)|breast(49)|central_nervous_sy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CCDC137</td>\n",
       "      <td>chr17</td>\n",
       "      <td>81671752</td>\n",
       "      <td>81671752</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>c.(505-507)aAg&gt;aTg</td>\n",
       "      <td>86.57</td>\n",
       "      <td>p.K169M</td>\n",
       "      <td>poly(A) RNA binding (GO:0044822)</td>\n",
       "      <td>biliary_tract(2)|central_nervous_system(45)|pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>UTS2R</td>\n",
       "      <td>chr17</td>\n",
       "      <td>82374762</td>\n",
       "      <td>82374762</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>c.(436-438)agC&gt;agG</td>\n",
       "      <td>112.01</td>\n",
       "      <td>p.S146R</td>\n",
       "      <td>G-protein coupled receptor activity (GO:000493...</td>\n",
       "      <td>biliary_tract(2)|breast(48)|central_nervous_sy...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol Chromosome  Start_Position  End_Position Variant_Classification  \\\n",
       "0       NLRP1      chr17         5541887       5541887      Missense_Mutation   \n",
       "1      FBXO39      chr17         6779878       6779878      Missense_Mutation   \n",
       "2        TP53      chr17         7674220       7674220      Missense_Mutation   \n",
       "3    ANKRD13B      chr17        29609163      29609163      Missense_Mutation   \n",
       "4        LIG3      chr17        34983021      34983021      Missense_Mutation   \n",
       "5       ABCA9      chr17        69016308      69016308      Missense_Mutation   \n",
       "6        GGA3      chr17        75239434      75239434      Missense_Mutation   \n",
       "7     CCDC137      chr17        81671752      81671752      Missense_Mutation   \n",
       "8       UTS2R      chr17        82374762      82374762      Missense_Mutation   \n",
       "\n",
       "  Variant_Type Reference_Allele Tumor_Seq_Allele2 Tumor_Seq_Allele1  \\\n",
       "0          SNP                C                 T                 C   \n",
       "1          SNP                G                 A                 G   \n",
       "2          SNP                C                 T                 C   \n",
       "3          SNP                G                 C                 G   \n",
       "4          SNP                A                 G                 A   \n",
       "5          SNP                A                 G                 A   \n",
       "6          SNP                G                 A                 G   \n",
       "7          SNP                A                 T                 A   \n",
       "8          SNP                C                 G                 C   \n",
       "\n",
       "           Codon_Change    TLOD Protein_Change  \\\n",
       "0  c.(2668-2670)aGa>aAa  209.99        p.R890K   \n",
       "1      c.(10-12)Gaa>Aaa   37.12          p.E4K   \n",
       "2    c.(742-744)cGg>cAg  264.57        p.R248Q   \n",
       "3    c.(643-645)Gac>Cac   22.18        p.D215H   \n",
       "4      c.(16-18)Aag>Gag   14.24          p.K6E   \n",
       "5  c.(2983-2985)cTt>cCt   49.96        p.L995P   \n",
       "6  c.(1504-1506)cCg>cTg  178.91        p.P502L   \n",
       "7    c.(505-507)aAg>aTg   86.57        p.K169M   \n",
       "8    c.(436-438)agC>agG  112.01        p.S146R   \n",
       "\n",
       "                               GO_Molecular_Function  \\\n",
       "0  ATP binding (GO:0005524)|cysteine-type endopep...   \n",
       "1                                                NaN   \n",
       "2  ATP binding (GO:0005524)|chaperone binding (GO...   \n",
       "3                                                NaN   \n",
       "4  ATP binding (GO:0005524)|DNA binding (GO:00036...   \n",
       "5  ATP binding (GO:0005524)|ATPase activity (GO:0...   \n",
       "6       ADP-ribosylation factor binding (GO:0030306)   \n",
       "7                   poly(A) RNA binding (GO:0044822)   \n",
       "8  G-protein coupled receptor activity (GO:000493...   \n",
       "\n",
       "                        COSMIC_tissue_types_affected  \n",
       "0  biliary_tract(2)|breast(32)|central_nervous_sy...  \n",
       "1  biliary_tract(2)|breast(11)|central_nervous_sy...  \n",
       "2  NS(1495)|adrenal_gland(724)|autonomic_ganglia(...  \n",
       "3  biliary_tract(2)|breast(11)|central_nervous_sy...  \n",
       "4  biliary_tract(2)|breast(223)|central_nervous_s...  \n",
       "5  biliary_tract(2)|breast(11)|central_nervous_sy...  \n",
       "6  biliary_tract(2)|breast(49)|central_nervous_sy...  \n",
       "7  biliary_tract(2)|central_nervous_system(45)|pa...  \n",
       "8  biliary_tract(2)|breast(48)|central_nervous_sy...  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "muts_selected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "57cc86d8-6dde-46f0-af4c-d9d8fedc6c90",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_muts = muts_selected[['Hugo_Symbol','Start_Position']]\n",
    "our_muts_results = final_muts.rename(columns={'Start_Position': 'Position'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "c2f0cd06-0631-489b-b48e-fc0c2c795bfb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NLRP1</td>\n",
       "      <td>5541887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FBXO39</td>\n",
       "      <td>6779878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7674220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ANKRD13B</td>\n",
       "      <td>29609163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIG3</td>\n",
       "      <td>34983021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>69016308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GGA3</td>\n",
       "      <td>75239434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CCDC137</td>\n",
       "      <td>81671752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>UTS2R</td>\n",
       "      <td>82374762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Position\n",
       "0       NLRP1   5541887\n",
       "1      FBXO39   6779878\n",
       "2        TP53   7674220\n",
       "3    ANKRD13B  29609163\n",
       "4        LIG3  34983021\n",
       "5       ABCA9  69016308\n",
       "6        GGA3  75239434\n",
       "7     CCDC137  81671752\n",
       "8       UTS2R  82374762"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "our_muts_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90297400-c8d1-4326-9248-bca6c344ec59",
   "metadata": {},
   "source": [
    "To assess the biological plausibility of the detected variants, the filtered mutation set was compared against an external curated mutation dataset for the HCC1143 cell line. Reference mutation data were restricted to chromosome 17."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "b783adc5-f4d3-4e23-af79-fc049cc6d8ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>Variant Type</th>\n",
       "      <th>Variant Info</th>\n",
       "      <th>Ref Allele</th>\n",
       "      <th>Alt Allele</th>\n",
       "      <th>Allele Fraction</th>\n",
       "      <th>Ref Count</th>\n",
       "      <th>Alt Count</th>\n",
       "      <th>...</th>\n",
       "      <th>NMD</th>\n",
       "      <th>Vep Somatic</th>\n",
       "      <th>Vep Impact</th>\n",
       "      <th>Oncogene High Impact</th>\n",
       "      <th>AM class</th>\n",
       "      <th>AM Pathogenicity</th>\n",
       "      <th>Hotspot</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Rescue Reason</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>NLRP1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>5541887</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.982</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0&amp;1</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.1477</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/17</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>FBXO39</td>\n",
       "      <td>chr17</td>\n",
       "      <td>6779878</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.384</td>\n",
       "      <td>32</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.1088</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>TP53</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7674220</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.982</td>\n",
       "      <td>0</td>\n",
       "      <td>55</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0&amp;1&amp;1&amp;1&amp;1&amp;1</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9963</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7/11</td>\n",
       "      <td>OncoKB, Cosmic, Hess</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>ANKRD13B</td>\n",
       "      <td>chr17</td>\n",
       "      <td>29609163</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.431</td>\n",
       "      <td>29</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.3163</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/15</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>LIG3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>34983021</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.204</td>\n",
       "      <td>43</td>\n",
       "      <td>10</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.1235</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/20</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>SRSF1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>58006457</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.170</td>\n",
       "      <td>71</td>\n",
       "      <td>14</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.9649</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2/4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>chr17</td>\n",
       "      <td>69016308</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.220</td>\n",
       "      <td>66</td>\n",
       "      <td>18</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>0.8596</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22/39</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>GGA3</td>\n",
       "      <td>chr17</td>\n",
       "      <td>75239434</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0&amp;1</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.0538</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14/17</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>CCDC137</td>\n",
       "      <td>chr17</td>\n",
       "      <td>81671752</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>0.402</td>\n",
       "      <td>55</td>\n",
       "      <td>38</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>0.1964</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4/6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>UTS2R</td>\n",
       "      <td>chr17</td>\n",
       "      <td>82374762</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>0.990</td>\n",
       "      <td>0</td>\n",
       "      <td>103</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0&amp;1&amp;1</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>False</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>0.5180</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3/3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 62 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Chromosome  Position Variant Type      Variant Info Ref Allele  \\\n",
       "70     NLRP1      chr17   5541887          SNV  missense_variant          C   \n",
       "71    FBXO39      chr17   6779878          SNV  missense_variant          G   \n",
       "72      TP53      chr17   7674220          SNV  missense_variant          C   \n",
       "73  ANKRD13B      chr17  29609163          SNV  missense_variant          G   \n",
       "74      LIG3      chr17  34983021          SNV  missense_variant          A   \n",
       "75     SRSF1      chr17  58006457          SNV  missense_variant          G   \n",
       "76     ABCA9      chr17  69016308          SNV  missense_variant          A   \n",
       "77      GGA3      chr17  75239434          SNV  missense_variant          G   \n",
       "78   CCDC137      chr17  81671752          SNV  missense_variant          A   \n",
       "79     UTS2R      chr17  82374762          SNV  missense_variant          C   \n",
       "\n",
       "   Alt Allele  Allele Fraction  Ref Count  Alt Count  ...  NMD  Vep Somatic  \\\n",
       "70          T            0.982          0         53  ...  NaN          0&1   \n",
       "71          A            0.384         32         20  ...  NaN            1   \n",
       "72          T            0.982          0         55  ...  NaN  0&1&1&1&1&1   \n",
       "73          C            0.431         29         22  ...  NaN          NaN   \n",
       "74          G            0.204         43         10  ...  NaN          NaN   \n",
       "75          A            0.170         71         14  ...  NaN            1   \n",
       "76          G            0.220         66         18  ...  NaN          NaN   \n",
       "77          A            0.989          0         91  ...  NaN          0&1   \n",
       "78          T            0.402         55         38  ...  NaN          NaN   \n",
       "79          G            0.990          0        103  ...  NaN        0&1&1   \n",
       "\n",
       "   Vep Impact Oncogene High Impact           AM class AM Pathogenicity  \\\n",
       "70   MODERATE                False      likely_benign           0.1477   \n",
       "71   MODERATE                False      likely_benign           0.1088   \n",
       "72   MODERATE                False  likely_pathogenic           0.9963   \n",
       "73   MODERATE                False      likely_benign           0.3163   \n",
       "74   MODERATE                False      likely_benign           0.1235   \n",
       "75   MODERATE                False  likely_pathogenic           0.9649   \n",
       "76   MODERATE                False  likely_pathogenic           0.8596   \n",
       "77   MODERATE                False      likely_benign           0.0538   \n",
       "78   MODERATE                False      likely_benign           0.1964   \n",
       "79   MODERATE                False          ambiguous           0.5180   \n",
       "\n",
       "   Hotspot  Intron   Exon         Rescue Reason  \n",
       "70   False     NaN   6/17                   NaN  \n",
       "71   False     NaN    2/4                   NaN  \n",
       "72    True     NaN   7/11  OncoKB, Cosmic, Hess  \n",
       "73   False     NaN   6/15                   NaN  \n",
       "74   False     NaN   2/20                   NaN  \n",
       "75   False     NaN    2/4                   NaN  \n",
       "76   False     NaN  22/39                   NaN  \n",
       "77   False     NaN  14/17                   NaN  \n",
       "78   False     NaN    4/6                   NaN  \n",
       "79   False     NaN    3/3                   NaN  \n",
       "\n",
       "[10 rows x 62 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "real_muts = pd.read_csv(\"HCC1143 mutations.csv\")\n",
    "real_muts=real_muts[real_muts['Chromosome']== 'chr17']\n",
    "real_muts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "e773c6d0-a204-4759-9c90-6286eeac955d",
   "metadata": {},
   "outputs": [],
   "source": [
    "real_muts_for_comparison= real_muts[['Gene','Position']]\n",
    "real_muts_for_comparison = real_muts_for_comparison.rename(columns={'Gene': 'Hugo_Symbol'})\n",
    "real_muts_for_comparison=real_muts_for_comparison.reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "741b5b61-1223-40c2-a69b-967e746fc631",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NLRP1</td>\n",
       "      <td>5541887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FBXO39</td>\n",
       "      <td>6779878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7674220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ANKRD13B</td>\n",
       "      <td>29609163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIG3</td>\n",
       "      <td>34983021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRSF1</td>\n",
       "      <td>58006457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>69016308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GGA3</td>\n",
       "      <td>75239434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CCDC137</td>\n",
       "      <td>81671752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>UTS2R</td>\n",
       "      <td>82374762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Position\n",
       "0       NLRP1   5541887\n",
       "1      FBXO39   6779878\n",
       "2        TP53   7674220\n",
       "3    ANKRD13B  29609163\n",
       "4        LIG3  34983021\n",
       "5       SRSF1  58006457\n",
       "6       ABCA9  69016308\n",
       "7        GGA3  75239434\n",
       "8     CCDC137  81671752\n",
       "9       UTS2R  82374762"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "real_muts_for_comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "28d5f1ee-0ecb-41ca-847b-bb954f20e03c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NLRP1</td>\n",
       "      <td>5541887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FBXO39</td>\n",
       "      <td>6779878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7674220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ANKRD13B</td>\n",
       "      <td>29609163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LIG3</td>\n",
       "      <td>34983021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ABCA9</td>\n",
       "      <td>69016308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GGA3</td>\n",
       "      <td>75239434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CCDC137</td>\n",
       "      <td>81671752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>UTS2R</td>\n",
       "      <td>82374762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Position\n",
       "0       NLRP1   5541887\n",
       "1      FBXO39   6779878\n",
       "2        TP53   7674220\n",
       "3    ANKRD13B  29609163\n",
       "4        LIG3  34983021\n",
       "5       ABCA9  69016308\n",
       "6        GGA3  75239434\n",
       "7     CCDC137  81671752\n",
       "8       UTS2R  82374762"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "our_muts_results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3d68947e-f95c-4daf-b95c-49dd7b521d76",
   "metadata": {},
   "source": [
    "## Conclusions and limitations\n",
    "\n",
    "Somatic variant calling on chromosome 17 identified **9/10** missense SNVs reported for the HCC1143 cell line in the DepMap reference dataset. All recovered variants matched the reference data exactly at genomic position level, demonstrating high concordance between the applied GATK-based workflow and curated mutation profiles for this cell line.\n",
    "\n",
    "Only one reference mutation, located in SRSF1, was not detected in the present analysis. The absence of this variant may be explained by multiple factors, including insufficient sequencing depth, low VAF due to subclonality or differences in cell line passage. Additionally, SRSF1 is located in a GC-rich genomic region, which can be challenging during library preparation and exome capture, leading to reduced amplification efficiency and decreased sensitivity for mutation detection at this locus. Notably, the missing SRSF1 mutation exhibits the lowest reported VAF in the reference real_muts dataset. \n",
    "\n",
    "## Possible improvements and future refinements\n",
    "\n",
    "Further sensitivity could potentially be achieved by incorporating a Panel of Normals (PoN) to improve removal of recurrent technical artifacts. While PoN usage may not directly rescue true low-coverage variants, it can reduce background noise and increase confidence in borderline calls. "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
